voltdb/datadog_checks/voltdb/config.py

```diff
@@ -0,0 +1,91 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+from typing import List, Optional, Tuple, Union
+
+import requests
+from six.moves.urllib.parse import urljoin, urlparse
+
+from datadog_checks.base import ConfigurationError, is_affirmative
+
+from .types import Instance
+
+
+class Config(object):
+    def __init__(self, instance):
+        # type: (Instance) -> None
+        url = instance.get('url')  # type: Optional[str]
+        username = instance.get('username')  # type: Optional[str]
+        password = instance.get('password')  # type: Optional[str]
+        password_hashed = is_affirmative(instance.get('password_hashed', False))  # type: bool
+        tags = instance.get('tags', [])  # type: List[str]
+
+        if not url:
+            raise ConfigurationError('url is required')
+
+        if username and not password:
+            raise ConfigurationError('password is required')
+
+        if password and not username:
+            raise ConfigurationError('username is required')
```

Comment:
Can this be simplified:

```suggestion
        if not username and not password:
            raise ConfigurationError('username and password is required')
```

Reply:
VoltDB allows exposing the HTTP API w/o auth credentials, so we can't really assume that `username` and `password` will always be set (which is also why `spec.yaml` doesn't mark them as `required`).